models/solution/route.rb

```diff
@@ -69,6 +97,16 @@ def shift_route_times(shift_amount, shift_start_index = 0)
             waiting_resorb = [stop.info.waiting_time - current_shift, 0].max
             current_shift = [current_shift - waiting_resorb, 0].max
             stop.info.waiting_time -= waiting_resorb
+          else
+            active_tw = stop.active_timewindow
+            if active_tw
+              current_shift =
+                if shift_amount >= 0
+                  [current_shift, active_tw.safe_end(stop.activity.lateness_allowed?) - stop.info.begin_time].min
```

Comment:
We cannot do this, this is a `raise` situation because we already shift the previous services X amount to the right but here you realise you cannot shift the current stop X amount. But then the end of the previous stop and the beginning of the current stop would overlap, right? Or, did I miss something.

Reply:
You are correct !

Reply:
Maybe we can pre-calculate how much actually we "can" shift and then shift everything that amount (i..e, we run the algo first without modifying the values and then use the final current_shift_amount to shift everyhign (and maybe raise a warning if the calculated maximum shift_amount is different than the original shift_amount) but I am not sure if we are on the right track with this function.